raiden/transfer/state.py

```diff
@@ -1504,26 +1511,28 @@ def to_dict(self) -> Dict[str, Any]:
 
     @classmethod
     def from_dict(cls, data: Dict[str, Any]) -> 'NettingChannelEndState':
-        onchain_locksroot = None
+        onchain_locksroot = EMPTY_MERKLE_ROOT
```

Comment:
@rakanalh You touched this code the last time and I think you were relying on this being None by default. Can you tell me why and what the thinking behind this was?

A `None` value can't be serialized as bytes so something is wrong here. And this is never supposed to be `None` since it's already initialized at construction time as you can see [here](https://github.com/raiden-network/raiden/pull/3800/files#diff-a6042df3a2b6bca66372154b6cceff89R1457).

I changed it temporarily to be initialized at the same value as in the constructor, but I guess the proper thing would be to remove the `if data['onchain_locksroot']` completely since in the the `onchain_locksroot` attribute should never be `None` and  `to_dict()` always serializes it.

Reply:
I think we can also go for the `EMPTY_MERKLE_ROOT` approach... i probably opted for None only because i forgot we had that constant defined. 

> I changed it temporarily to be initialized at the same value as in the constructor, but I guess the proper thing would be to remove the if data['onchain_locksroot'] completely since in the the onchain_locksroot attribute should never be None and to_dict() always serializes it.

Sounds good to me

Reply:
Just occured to me that we might want to do this in a separate migration because:
1. If this is merged after a release, we could end up having people with `None` as the value for that field.
2. If we don't, then we have to tell people who run master to go back to earlier databases to trigger Raiden to re-run some migrations which put the `EMPTY_MERKLE_ROOT` instead of None into the DB

Reply:
Hmm what I am asking is how can `None` actually end up there? Whenever a `NettingChannelEndState` is initialized the `onchain_locksroot` attribute is initialized to `EMPTY_MERKLE_ROOT`. 

So no locksroot seems to ever be initialized with `None`, apart from there where I removed it at `from_dict()`. But `from_dict()` will always deserialize existing `NettingChannelEndState` which should always have had the attribute initialized.

You added the attribute in [this](https://github.com/raiden-network/raiden/commit/3cb8e2b632ebd9021d01ebdf9dcf9d1bb3d133a1) commit to the `NettingChannelEndState`.

And the corresponding DB migration seems to never set it to None:

https://github.com/raiden-network/raiden/blob/8797ec58a776d992b18ecd26f7788926546b2eac/raiden/storage/migrations/v19_to_v20.py#L174-L182

Reply:
Oh i see your point. So in the constructor, i am correctly initializing it to `EMPTY_MERKLE_ROOT`, while in serializing / deserializing i am using None :man_facepalming: 

> Hmm what I am asking is how can None actually end up there?

It shouldn't... my bad. 

> So no locksroot seems to ever be initialized with None, apart from there where I removed it at from_dict(). But from_dict() will always deserialize existing NettingChannelEndState which should always have had the attribute initialized.
> You added the attribute in this commit to the NettingChannelEndState.
> And the corresponding DB migration seems to never set it to None:

I actually do it here:
https://github.com/raiden-network/raiden/blob/8797ec58a776d992b18ecd26f7788926546b2eac/raiden/storage/migrations/v19_to_v20.py#L85-L86

So since we have a release containing this, we should fix those in a separate migration

Reply:
PS: i can help do this in a separate PR

Reply:
As discussed in the office, the change is done here: https://github.com/raiden-network/raiden/pull/3800/commits/82294cfb263c94b841bded00c2f2c6dcca71a2b0

A separate migration won't be neded as long as we release a new RC right now since this has not yet been included in an actual release.